src/dft/backends/mklgpu/forward.cpp

```diff
@@ -0,0 +1,193 @@
+/*******************************************************************************
+* Copyright Codeplay Software Ltd.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions
+* and limitations under the License.
+*
+*
+* SPDX-License-Identifier: Apache-2.0
+*******************************************************************************/
+
+#include <type_traits>
+#if __has_include(<sycl/sycl.hpp>)
+#include <sycl/sycl.hpp>
+#else
+#include <CL/sycl.hpp>
+#endif
+
+#include "oneapi/mkl/types.hpp"
+#include "oneapi/mkl/exceptions.hpp"
+
+#include "oneapi/mkl/dft/detail/mklgpu/onemkl_dft_mklgpu.hpp"
+#include "oneapi/mkl/dft/detail/types_impl.hpp"
+#include "oneapi/mkl/dft/detail/descriptor_impl.hpp"
+
+#include "dft/backends/mklgpu/mklgpu_helpers.hpp"
+
+// MKLGPU header
+#include "oneapi/mkl/dfti.hpp"
+
+/**
+Note that in this file, the Intel MKL-GPU library's interface mirrors the interface
+of this OneMKL library. Consequently, the types under dft::TYPE are closed-source MKL types, 
+and types under dft::detail::TYPE are from this library.
+**/
+
+namespace oneapi {
+namespace mkl {
+namespace dft {
+namespace mklgpu {
+namespace detail {
+/// Forward a MKLGPU DFT call to the backend, checking that the commit impl is valid.
+/// Assumes backend descriptor values match those of the frontend.
+template <dft::detail::precision prec, dft::detail::domain dom, typename... ArgTs>
+inline auto compute_forward(dft::detail::descriptor<prec, dom> &desc, ArgTs &&... args) {
+    using mklgpu_desc_t = dft::descriptor<to_mklgpu(prec), to_mklgpu(dom)>;
+    using commit_t = dft::detail::commit_impl;
```

Comment:
this seems unused

Reply:
yes. removed in [92b7fa10937442ea22d7a149cf4afa7312623187](https://github.com/oneapi-src/oneMKL/pull/259/commits/92b7fa10937442ea22d7a149cf4afa7312623187)